Source/DafnyCore/Backends/DafnyExecutableBackend.cs

```diff
@@ -11,14 +11,17 @@ namespace Microsoft.Dafny.Compilers;
 
 public abstract class DafnyExecutableBackend : ExecutableBackend {
 
+  protected virtual bool PreventShadowing => true;
+  protected virtual bool SupportsUncompilableCode => true;
```

Comment:
nit: reads like "can compile uncompilable code" but I think we want a shorter version of "supports outputting partial code for programs that have uncompilable code"

Reply:
Renamed to `CanEmitUncompilableCode`